### ISSUE TITLE
"index with mask" failure of TensorRT 8.6 when running trtexec on GPU A4500

### ISSUE THREAD

**Issue by FrankyTang** [NVIDIA/TensorRT#4010]
## Description
I tried to get tensor with index，codes is 
""" 
class SimpleModel(nn.Module):
    def forward(self, x, mask):
        out = x[mask]
        return out
"""
The onnx graph is 
![image](https://github.com/user-attachments/assets/813db29e-24ba-4d29-9be1-5ac7a72c7cb4)

A bug from onnx to trt
![image](https://github.com/user-attachments/assets/9f8bf87b-7c89-47dd-a25e-8f90aa1a38da)

## Environment

docker is :nvcr.io/nvidia/pytorch:23.04-py3

## Relevant Files

**Model link**:
[test.tar.gz](https://github.com/user-attachments/files/16201560/test.tar.gz)

## Steps To Reproduce
trtexec --onnx=simp_model_sim.onnx --saveEngine=tmp.trt

**Commands or scripts**:

**Have you tried [the latest release](https://developer.nvidia.com/tensorrt)?**:
My environment does not support using more advanced versions.

**Can this model run on other frameworks?** For example run ONNX model with ONNXRuntime (`polygraphy run <model.onnx> --onnxrt`): 
onnxrt is PASSED.

**Comment by lix19937** [comment]
I think it related to `data-depended shape`.

**Comment by FrankyTang** [comment]
@lix19937 My tensorrt version is 8.6.1, I think it supported data-depended shape, like non_zero, where

**Comment by lix19937** [comment]
You can try the trt latest version.

**Comment by poweiw** [comment]
Hello there's a sample for data-dependent shape plugins [here](https://github.com/NVIDIA/TensorRT/tree/release/10.8/samples/sampleNonZeroPlugin). You can also override [IOutputAllocator](https://docs.nvidia.com/deeplearning/tensorrt/api/c_api/classnvinfer1_1_1v__1__0_1_1_i_output_allocator.html) for this purpose. 
Closing for now but feel free to reopen if you still see the issue for trt >=10.0!
